rasa/nlu/classifiers/diet_classifier.py

```diff
@@ -1252,6 +1258,193 @@ def __init__(
 
         self._prepare_layers()
 
+    def adjust_sparse_layers(
+        self,
+        data_example: Dict[Text, List[FeatureArray]],
+        new_feature_sizes: Dict[Text, Dict[Text, List[int]]],
+        old_feature_sizes: Dict[Text, Dict[Text, List[int]]],
+    ) -> None:
+        """Adjusts sizes of DenseForSparse layers.
+
+        Updates sizes of DenseForSparse layers by comparing current sparse feature
+        sizes to old ones. This must be done before fine-tuning starts to account for
+        sparse features that might have changed. If any layer is changed, compiles
+        the model and fits a sample data on it to activate updated layer(s).
+
+        Args:
+            data_example: a data example that is stored in DIETClassifier class.
+            new_feature_sizes: sizes of current sparse features.
+            old_feature_sizes: sizes of sparse features the model was trained on before.
+        """
+        model_changed = False
+        for attribute in new_feature_sizes:
+            layer_dicts = self._get_dense_for_sparse_layers(attribute)
+            for layer_dict in layer_dicts:
+                layer_changed = self._update_dense_for_sparse_layer(
+                    layer_dict=layer_dict,
+                    new_feature_sizes=new_feature_sizes[attribute],
+                    old_feature_sizes=old_feature_sizes[attribute],
+                )
+                if layer_changed:
+                    model_changed = True
+        if model_changed:
+            self._compile_and_fit(data_example)
+
+    def _update_dense_for_sparse_layer(
```

Comment:
A function name is always a verb which tells what the function is doing. From what I see the function is checking whether the layer has been updated or not. It's not the same as "updating" the layer.